Framework/Algorithms/test/ConvertSpectrumAxis2Test.h

```diff
@@ -391,4 +391,34 @@ class ConvertSpectrumAxis2Test : public CxxTest::TestSuite {
   }
 };
 
+class ConvertSpectrumAxis2TestPerformance : public CxxTest::TestSuite {
+public:
+  static ConvertSpectrumAxis2TestPerformance *createSuite() {
+    return new ConvertSpectrumAxis2TestPerformance();
+  }
+  static void destroySuite(ConvertSpectrumAxis2TestPerformance *suite) {
+    delete suite;
+  }
+
+  ConvertSpectrumAxis2TestPerformance() {}
+
+  void test_conversion_to_signed_theta_with_many_entries() {
+    std::string inputWSName("inputWS");
+    std::string outputWSName("outputWS");
+
+    auto testWS = WorkspaceCreationHelper::create2DWorkspaceWithFullInstrument(
+        20000, 20000);
+    AnalysisDataService::Instance().addOrReplace(inputWSName, testWS);
+
+    Mantid::Algorithms::ConvertSpectrumAxis2 conv;
+
+    conv.initialize();
```

Comment:
if you had `conv.setChild(True)`. You probably wouldn't need to do that nonsense with the ADS that everyone seems to like.